### PR TITLE
[New] Webhook retry enhancement docs

### PR DESCRIPTION
**Pull request by shikhartanwar**
## Description

This PR makes the following changes -
 - Adding docs incorporating webhook retry improvements
    - Change in retry policy (retried upto 24h now)
    - Introduction of unstable status
    - Captured corner-case/outlier-scenario when an event is dropped
    - Use consistent terminology for event registration
  - Re-emphasized importance of signature verification for webhook deliveries
  - Added custom events page which was missing from this repo
  - Fixed many broken links:
     - For webhook and journaling intro pages
     - In the FAQ page

## Related Issue
https://github.com/AdobeDocs/adobeio-events/pull/87

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by sangeetha5491** [comment]
Can we add screenshots for src/pages/guides/img/disabled-status.png and unstable status from Prod? because the journalling API shows "stage".

File: src/pages/guides/using/custom_events.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Custom I/O Events Overview
```
Comment (sangeetha5491) [inline src/pages/guides/using/custom_events.md:1]: need to add this to gatsby-config.js

Comment (shikhartanwar) [reply src/pages/guides/using/custom_events.md:1]: Good catch. Will do that!

Comment (shikhartanwar) [reply src/pages/guides/using/custom_events.md:1]: Done. Btw I just realized that adding the pages is only required for side and global navigation. They will get rendered if they are linked from somewhere else.

Comment (sangeetha5491) [reply src/pages/guides/using/custom_events.md:1]: yes, i felt we do need it to be part of the side menu.

**Comment by shikhartanwar** [comment]
> Can we add screenshots for src/pages/guides/img/disabled-status.png and unstable status from Prod? because the journalling API shows "stage".

@sangeetha5491 This is also done.

File: gatsby-config.js [hunk 2]
```diff
@@ -125,7 +125,7 @@ module.exports = {
           },
           {
             title: 'Cloud Manager Events',
-            path: '/experience-cloud/cloud-manager/guides/getting-started/create-event-integration/'
+            path: 'https://www.adobe.io/apis/experiencecloud/cloud-manager/docs.html#!AdobeDocs/cloudmanager-api-docs/master/create-event-integration.md'
```
Comment (icaraps) [inline gatsby-config.js:128]: Why was this path changed to the old URL which will be decommissioned soon ?

**Comment by icaraps** [comment]
Please don't hardcode URLs with `https://www.adobe.io` or else the staging links will point to the production links which we want to avoid.

File: gatsby-config.js (hunk 2 above)
Comment (shikhartanwar) [reply gatsby-config.js:128]: @icaraps Ok but this relative path does not exist in this repo, so how will it get resolved?

**Comment by shikhartanwar** [comment]
> Please refrain from hardcoding URLs with `https://www.adobe.io` or else the staging pages will have links that are pointing to production which we want to avoid.

Got it, but how do links to pages that are not in this repo get resolved.

File: gatsby-config.js (hunk 2 above)
Comment (icaraps) [reply gatsby-config.js:128]: It's resolved once it's on stage or prod

**Comment by icaraps** [comment]
It's resolved once it's on stage or prod

**Comment by icaraps** [comment]
All you need is just to put the URL pathname for example if you want to link to the url `https://developer-stage.adobe.com/events/docs/guides/journaling_intro/` and you want it to work on stage and prod from anywhere, remove the origin from the url so it turns into ` /events/docs/guides/journaling_intro/`

Or programmatically, you can do `new URL(url).pathname`to find out the correct url for the link before writing it in the markdown file.

And if you want to find out the url for a given repo, you can check the pathPrefix eg https://github.com/AdobeDocs/adobe-io-runtime/blob/main/gatsby-config.js#L214

File: gatsby-config.js (hunk 2 above)
Comment (shikhartanwar) [reply gatsby-config.js:128]: Ok. I have to push another documentation related PR next week so i'll take care of this.

**Comment by shikhartanwar** [comment]
> All you need is just to put the URL pathname for example if you want to link to the url `https://developer-stage.adobe.com/events/docs/guides/journaling_intro/` and you want it to work on stage and prod from anywhere, remove the origin from the url so it turns into ` /events/docs/guides/journaling_intro/`
> 
> Or programmatically, you can do `new URL(url).pathname`to find out the correct url for the link before writing it in the markdown file.
> 
> And if you want to find out the url for a given repo, you can check the pathPrefix eg https://github.com/AdobeDocs/adobe-io-runtime/blob/main/gatsby-config.js#L214

Any way to get such links working in dev environment as well?

**Comment by icaraps** [comment]
You can try to create a proxy in your dev env see https://www.gatsbyjs.com/docs/api-proxy/ 

e.g. add the following in your gatsby-config.js: 

```
proxy: {
  prefix: '/apis',
  url: 'https://www.adobe.io'
},
```

I didn't try it myself, let me know how it goes.

**Comment by shikhartanwar** [comment]
> You can try to create a proxy in your dev env see https://www.gatsbyjs.com/docs/api-proxy/
> 
> e.g. add the following in your gatsby-config.js:
> 
> ```
> proxy: {
>   prefix: '/events/docs/guides/journaling_intro',
>   url: 'https://developer-stage.adobe.com'
> },
> ```
> 
> I didn't try it myself, let me know how it goes. You can add multiple proxies.

Thank you @icaraps. I was able to get this working but since there were multiple such pages, I had to create an array of proxy objects:
```
  proxy: [
    {
      prefix: '/experience-cloud/cloud-manager',
      url: 'https://developer-stage.adobe.com'
    },
    {
      prefix: '/app-builder',
      url: 'https://developer-stage.adobe.com'
    },
    {
      prefix: '/console/docs/guides',
      url: 'https://developer-stage.adobe.com'
    },
    {
      prefix: '/authentication',
      url: 'https://developer.adobe.com'
    },
  ],
```

**Comment by icaraps** [comment]
> I was able to get this working but since there were multiple such pages, I had to create an array of proxy objects

That's great ! If you have time, could you pls share your findings on the devsite slack channel ? I believe other folks could make use of it.

**Comment by shikhartanwar** [comment]
> > I was able to get this working but since there were multiple such pages, I had to create an array of proxy objects
> 
> That's great ! If you have time, could you pls share your findings on the devsite slack channel ? I believe other folks could make use of it.

Sure I can, but unfortunately not all links are working perfectly. The one's that are pointing to the older www.adobe.io domain (`/apis/experienceplatform` etc.) are not getting rendered mostly because of a **CORS** issue.

**Comment by icaraps** [comment]
cc @timkim maybe this is something that can be handled in the Fastly config

**Pull request merged**